controller_interface/include/controller_interface/controller_interface_base.hpp

```diff
@@ -153,6 +155,21 @@ class ControllerInterfaceBase : public rclcpp_lifecycle::node_interfaces::Lifecy
   CONTROLLER_INTERFACE_PUBLIC
   virtual return_type update(const rclcpp::Time & time, const rclcpp::Duration & period) = 0;
 
+  /**
+   * Trigger update method. This method is used by the controller_manager to trigger the update
+   * method of the controller.
+   * The method is used to trigger the update method of the controller synchronously or
+   * asynchronously, based on the controller configuration.
+   * **The method called in the (real-time) control loop.**
+   *
+   * \param[in] time The time at the start of this control loop iteration
+   * \param[in] period The measured time taken by the last control loop iteration
+   * \returns return_type::OK if update is successfully, otherwise return_type::ERROR.
```

Comment:
the boolean value should also be documented here

Reply:
OK

Reply:
```suggestion
   * \returns A pair with the first element being a boolean indicating if the async callback method was triggered and the second element being the last return value of the async function. For more details check the AsyncFunctionHandler implementation in `realtime_tools` package.
```